lbz/authz/decorators.py

```diff
@@ -18,7 +18,9 @@ def check_permission(resource: Resource, permission_name: str) -> dict:
     Raises if not.
     """
     authorization_header = resource.request.headers.get("Authorization", "")
-    guest_authorization_policy = resource.get_guest_authorization()
+    guest_authorization_policy = (
+        resource.get_guest_authorization() if not authorization_header else None
```

Comment:
I liked [the version before introducing this bug](https://github.com/pdyba/lambdalizator/blob/0ffb9f4187d7b6c5fb1935ab17858df49ca0de11/lbz/authz.py#L177) more. It is easier to read when we don't check the same thing multiple times (a little bit updated below):
```python
guest_authorization_policy = None
if not (authorization_header := resource.request.headers.get("Authorization", "")):
    if not (guest_authorization_policy := resource.get_guest_authorization()):
        raise Unauthorized("Authorization header missing or empty")
```

Reply:
What about the tests for this change? All of them are passing as before which doesn't mean it is good because we can introduce a similar bug in the future 😐